scipy/optimize/_lsq/lsq_linear.py

```diff
@@ -285,7 +283,13 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
     if b.size != m:
         raise ValueError("Inconsistent shapes between `A` and `b`.")
 
-    lb, ub = prepare_bounds(bounds, n)
+    if isinstance(bounds, Bounds):
+        lb = bounds.lb
+        ub = bounds.ub
+    else:
+        if len(bounds) != 2:
+            raise ValueError("`bounds` must contain 2 elements.")
+        lb, ub = prepare_bounds(bounds, n)
```

Comment:
What do you think about moving this logic to `prepare_bounds` since the purpose of this block is to prepare the bounds (for use by `lsq_linear`)?

Reply:
done

Reply:
I really mean all of this, but I'll go ahead and merge as-is if tests pass.